workers/price-reporter/src/reporter.rs

```diff
@@ -300,26 +244,73 @@ impl ConnectionMuxer {
     /// `StreamMap` for multiplexing
     async fn initialize_connections<'a>(
```

Comment:
I find it odd that the `ConnectionMuxer` handles price conversion. It's job is to manage connections and forward data, I feel the layers on top of this should do price conversion. 

It makes sense to do price conversion where we do aggregation; neither of which belong in the connection muxer.

Reply:
Plus putting it in the connection muxer means we subscribe to the `quote/default_quote` topic for _every_ new asset. Lots of wasted work + bandwidth

Reply:
The latter is a good point, currently all the default pairs are on the same quote, and this will generally be true...

I'll take a stab at having the `Reporter` (where price conversion currently occurs via `get_state`) be responsible for spawning multiple `ConnectionMuxer`s, maintaining the functionality that each `ConnectionMuxer` streams the same pair directly from all of the exchanges that support it, as opposed to spawning any connections necessary for reporting the price for a requested pair